05.ls/ls.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+COLUMN = 3
+
+def main
+  ready_files_to_display.each do |files|
+    files.each do |file|
+      print file.ljust(count_filname_char + 1) if file
+    end
+    print "\n"
+  end
+end
+
+def load_files
+  Dir.entries('.').sort.grep_v(/^\./)
+end
+
+def count_filname_char
+  load_files.max_by(&:length).length
```

Comment:
```suggestion
def count_filname_char
  sleep 1
  load_files.max_by(&:length).length
```

みたいにスリープを挟んだりするとかなり遅くなると思います。
処理効率を良くする方法を考えてみましょう。

一般論として、最近のコンピュータはめちゃ速いので、多少効率の悪いコードを書いても遅さを体感できなかったりしますが、簡単な変更で処理効率を上げられるのであれば、効率の良いロジックを選択するのがベターです。

Reply:
自分なりに解決してみました。
少し言葉にするのが難しいのですが、「メソッドは最後の式を戻り値として返す」というのは、実際には「メソッドの処理を最初から最後まで行い、結果として一番最後の式を戻り値として返す」という事象が、メソッドの戻り値を呼び出すときに起こっているのかなと考えてみました。

今回の例でいうと、mainメソッドでの
```ruby
files.each do |file|
  print file.ljust(count_filname_char + 1) if file
end
```
の際、毎回
```ruby
def count_filname_char
  sleep 1
  load_files.max_by(&:length).length
```
の処理を行ってしまうのが遅延の原因かな？と思います。

そのため、一度この値を変数に入れ、ループ内で毎回処理が行われないようにしてみました。

Reply:
また、
````ruby
main(files, max_char)
````
のように引数として渡す場合でも、今回の修正と同じく「一旦変数に入れる」処理になるので、遅延がしなくなるのかな？と思いました。

今回はmainメソッドに引数があるのが、あまりしっくり来なかったので、メソッド内で変数に代入をしました。